flink-table/flink-table-common/src/main/java/org/apache/flink/table/types/inference/strategies/ArrayComparableElementTypeStrategy.java

```diff
@@ -67,6 +67,9 @@ public ArgumentCount getArgumentCount() {
     public Optional<List<DataType>> inferInputTypes(
             CallContext callContext, boolean throwOnFailure) {
         final List<DataType> argumentDataTypes = callContext.getArgumentDataTypes();
+        if (argumentDataTypes.size() < 1) {
+            return Optional.empty();
+        }
```

Comment:
why do we need that? That should be covered alrready by the `ArgumentCount`

Reply:
Yes, we need it. In this situation -> SELECT ARRAY_MIN(), although we set argumentCount to one, but argumentDataTypes.size() == 0, if we do not check whether it is smaller than one, it report out of boundary exception.

But if the input arguments number is larger than one, argumentCount will handle this situation.

Reply:
@dawidwys Is this OK?
I think we can use 
` if (argumentDataTypes.size() < 1) {
            return Optional.empty();
        }`
first to handle empty input situation, because  Calcite's stack inference logic and the argument count check can work except empty argument input. Once the bug in the Calcite's stack inference logic and the argument count check solved, we can remove this if statement.
https://github.com/apache/flink/pull/22922/files#r1252434610

Reply:
I think we can use 
` if (argumentDataTypes.size() < 1) {
            return Optional.empty();
        }`
first to handle empty input situation, because  Calcite's stack inference logic and the argument count check can work except empty argument input. Once the bug in the Calcite's stack inference logic and the argument count check solved, we can remove this if statement.

Reply:
I find use argumentTypeStrategy to achieve this strategy can solve this problem. We can use argumentTypeStrategy to achieve this strategy.

Reply:
The change in `TypeInferenceOperandInference` has been implemented. @hanyuzheng7 Would you like to revert the changes around checking operand count? If not, I am happy to do that myself, we could merge the PR afterwards.